fodselsnummer.py

```diff
@@ -27,24 +27,54 @@ def check_fnr(fnr, d_numbers=True):
     if not FNR_REGEX.match(fnr):
         return False
 
-    try:
-        datetime.strptime(fnr[0:6], '%d%m%y')
-    except ValueError:
-        if d_numbers:
-            dnrdatestring = str(int(fnr[0])-4) + fnr[1:6]
-            try:
-                datetime.strptime(dnrdatestring, '%d%m%y')
-            except ValueError:
-                return False
+    individual_number = int(fnr[6:9])
+    day, month, year = int(fnr[0:2]), int(fnr[2:4]), int(fnr[4:6])
+    if 41 <= day <= 71:  # if D-number
+        if not d_numbers:
+            logger('Day out of range in fnr')
+            return False
+        day -= 40
+
+    if not (1 <= day <= 31):
+        logger('Day out of range in fnr')
+        return False
+    if 41 <= month <= 52:  # if H-number
+        if h_numbers:
+            month -= 40
         else:
+            logger('Month out of range')
             return False
+    if not 1 <= month <= 12:
+        logger('Month out of range in fnr')
```

Comment:
Consider using "fodselsnumber" or "personal number" or similar instead of "fnr" to make it more readable.